frontend/src/RecordingPageComp/SimilarSounds/SimilarSounds.jsx

```diff
@@ -9,20 +9,23 @@ function SimilarSounds({ playbackDuration }) {
   const [isLoadingSounds, setIsLoadingSounds] = useState(false);
 
   const fetchSounds = async () => {
-    setIsLoadingSounds(true); // Start loading
-    // Simulate loading time or wait for an actual event that confirms the instrument is ready
-    setTimeout(() => {
-      setIsLoadingSounds(false); // Stop loading after 1 second
-    }, 2000);
+    //Freesound API request
     try {
+      //Loading State
+      setIsLoadingSounds(true);
       const API_KEY = import.meta.env.VITE_APP_API_KEY;
       const response = await fetch(
         `https://freesound.org/apiv2/search/text/?token=${API_KEY}&filter=duration:${duration}&fields=name,previews`
       );
+      if (!response.ok) {
+        throw new Error(`HTTP error! Status: ${response.status}`);
```

Comment:
You may want to set loading state to false here too or in the catch block. Execution of the try block will stop when an error is thrown and continue in the catch block. So, the loading state will stay set to True when an error occurs. As we discussed offline, I would make sure to test all code paths to ensure behavior is expected in both success and failure cases.

Reply:
Will work on this when performing testing out paths that lead to an error.